app/main.py

```diff
@@ -1,3 +1,15 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrap_cache(*args):
+        stored_result = cache_dict.get(args, False)
+        if args in cache_dict:
+            print("Getting from cache")
+            return stored_result
+        else:
+            print("Calculating new result")
+            func_result = func(*args)
+            cache_dict[args] = func_result
+            return func_result
```

Comment:
You can use a single return statement outside of the if/else statement instead of two. Just get value from the dict.